File: golang/set1/challenge7/main.go
package main

import (
	"crypto/aes"
	"encoding/base64"
	"fmt"
	"io/ioutil"
)

func getBlocks(ciphertext []byte, blockSize int) [][]byte {
	blocks := [][]byte{}
	for i := 0; i < len(ciphertext); i += blockSize {
		if i+blockSize > len(ciphertext) {
			blocks = append(blocks, ciphertext[i:])
		} else {
			blocks = append(blocks, ciphertext[i:i+blockSize])
		}
	}
	return blocks
}

func ecbDecrypt(input []byte, key []byte) ([]byte, error) {
	output := []byte{}
	blocks := getBlocks(input, len(key))
	cipher, err := aes.NewCipher([]byte(key))
	if err != nil {
		return nil, err
	}
	for _, block := range blocks {
		dst := make([]byte, len(key))
		cipher.Decrypt(dst, block)
		output = append(output, dst...)
	}
	return output, nil
}

func main() {
	b, err := ioutil.ReadFile("7.txt")
	if err != nil {
		panic(err)
	}
	ciphertext, err := base64.StdEncoding.DecodeString(string(b))
	if err != nil {
		panic(err)
	}
	key := "YELLOW SUBMARINE"
	output, err := ecbDecrypt(ciphertext, []byte(key))
	if err != nil {
		panic(err)
	}
	fmt.Println(string(output))
}

File: rust/set1/challenge7/src/main.rs

extern crate rustc_serialize as serialize;
extern crate crypto;

use crypto::symmetriccipher::BlockDecryptor;
use crypto::aessafe::AesSafe128Decryptor;
use serialize::base64::{FromBase64};
use std::io::prelude::*;
use std::fs::File;

fn get_blocks<'a>(ciphertext: &'a [u8], block_size: u16) -> Vec<&'a [u8]> {
    let mut blocks: Vec<&'a [u8]> = vec![];
    let mut i = 0;
    for j in 0..ciphertext.len() {
        if i >= ciphertext.len() {
            break
        }
        if i + block_size as usize > ciphertext.len() {
            blocks.push(&ciphertext[i ..]);
        } else {
            blocks.push(&ciphertext[i .. i + block_size as usize]);
        }
        i += block_size as usize;
    }
    return blocks;
}

fn main() {
    let mut fh = File::open("7.txt").unwrap();
    let mut content: Vec<u8> = Vec::new();
    fh.read_to_end(&mut content).unwrap();
    let ciphertext = content.from_base64().unwrap();
    let key_size = 16;
    let key_str = "YELLOW SUBMARINE";
    let mut cipher = AesSafe128Decryptor::new(key_str.as_ref());
    let blocks = get_blocks(ciphertext.as_ref(), key_size);
    for block in blocks {
        let mut out_vec = vec![0; block.len()];
        cipher.decrypt_block(block, out_vec.as_mut_slice());
        let out_str = String::from_utf8_lossy(out_vec.as_ref());
        print!("{}", out_str);
    }
}
